File: backend/src/main/java/org/example/backend/models/ApiError.java
package org.example.backend.models;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;

@Data
public class ApiError {

    private int status;
    private String message;
    private LocalDateTime timestamp;
    private Map<String, Object> details = new HashMap<>();

    public void addDetail(String key, Object value) {
        details.put(key, value);
    }
}

File: frontend/src/types/ApiError.ts
import {isAxiosError} from "axios";

export interface ApiError {
    status: number,
    message: string,
    details?: { [key: string]: object };
}

export const toApiError = (e: unknown) => {
    if (!isAxiosError(e) || !e.response) {
        return null;
    }

    return e.response.data as ApiError;
};
